p2p/node/discover.go

```diff
@@ -0,0 +1,51 @@
+package node
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/ipfs/go-cid"
+	nodeSym "github.com/taubyte/go-sdk-symbols/p2p/node"
+	"github.com/taubyte/go-sdk/utils/codec"
+)
+
+func Discover(max uint32, timeout time.Duration) ([]cid.Cid, error) {
+	if max == 0 {
+		return []cid.Cid{}, nil
+	}
+
+	var (
+		id        uint32
+		peersSize uint32
+	)
+
+	err0 := nodeSym.DiscoverPeersSize(max, uint32(timeout), &id, &peersSize)
+	if err0 != 0 {
+		return nil, fmt.Errorf("discover failed with: %s", err0)
+	}
+	if peersSize == 0 {
+		return []cid.Cid{}, nil
+	}
+
+	peerBytes := make([]byte, peersSize)
+	err0 = nodeSym.DiscoverPeers(id, &peerBytes[0])
+	if err0 != 0 {
+		return nil, fmt.Errorf("reading peers after discover failed with: %s", err0)
+	}
+
+	var peerBytesSlice [][]byte
+	err := codec.Convert(peerBytes).To(&peerBytesSlice)
+	if err != nil || len(peerBytesSlice) == 0 {
```

Comment:
Maybe check len of peerBytes above first before trying to convert to confirm there is data to convert

Reply:
peerBytes is made from [peerSize](https://github.com/taubyte/go-sdk/pull/26/files/ebb4b230bfa5e439f36112e418c47f46ed03cf1c#diff-85f34a354509ed6101673e5c966e7e57a8a3daf761f67b3751146d7554f673e8R30) which has length > 0.

Reply:
Ah true